ert_shared/ensemble_evaluator/evaluator.py

```diff
@@ -234,20 +237,37 @@ def stop(self):
         self._ws_thread.join()
 
     def run_and_get_successful_realizations(self):
-        try:
-            with self.run() as mon:
-                for _ in mon.track():
-                    pass
-        except ConnectionRefusedError as e:
-            logger.debug(
-                f"run_and_get_successful_realizations caught {e}, cancelling or stopping ensemble..."
-            )
-            if self._ensemble.is_cancellable():
-                self._ensemble.cancel()
-            else:
-                self._stop()
-            self._ws_thread.join()
-            raise
+        monitor_context = self.run()
+        unsuccessfull_connection_attempts = 0
+        while True:
+            try:
+                with monitor_context as mon:
+                    for _ in mon.track():
+                        unsuccessfull_connection_attempts = 0
+                break
+            except ConnectionClosedError as e:
+                logger.debug(
```

Comment:
Just want to be sure here, is this the correct behaviour to always reconnect after the connection was lost?

Reply:
Yes, because it means we come from a connection that was working, and then we should always retry. If we are unable to connect we will hit the other exception path.